tests/integration/utils.py

```diff
@@ -53,3 +54,56 @@ def create_new_request(self, payload):
         )
         resp.raise_for_status()
         return Response(resp.json(), resp.json()["id"], resp.status_code)
+
+    def download_bundle(self, request_id):
+        """
+        Download a bundle archive.
+
+        :param int request_id:  ID of the request in Cachito
+        :return: Object that contains response from the Cachito API
+        :rtype: Response
+        :raises requests.exceptions.HTTPError: if the request to the Cachito API fails
+        """
+        resp = requests.get(f"{self._cachito_api_url}/requests/{request_id}/download")
+        resp.raise_for_status()
+        return Response(resp.content, request_id, resp.status_code)
+
+    def wait_for_complete_request(self, response):
+        """
+        Wait for a request to complete fetching the application source and dependencies.
+
+        :param Response response: Object that contains response from the Cachito API
+        :return: Object that contains response from the Cachito API
+        :rtype: Response
+        :raises TimeoutError: if the request would not complete within 5 minutes
+        """
+        start_time = time.time()
+        while response.data["state"] == "in_progress":
+            # 300 seconds
+            if time.time() - start_time >= 300:
+                raise TimeoutError("The Cachito request did not complete within 5 minutes")
+            time.sleep(5)
+            response = self.fetch_request(response.id)
+
+        return response
+
+
+def escape_path_go(dependency):
```

Comment:
Could you please explain why this is needed? This looks very interesting.

Reply:
I noticed that in the bundle the paths for dependencies changes when there is an uppercase letter. Escaping of paths is described in https://github.com/golang/mod/blob/2addee1ccfb22349ab47953a3046338e461eb4d1/module/module.go#L46
I hope it's relevant for all go modules.

Reply:
Could you add this link to the docstring? It was very helpful.